packages/odk/test/Adaptor.test.js

```diff
@@ -1,216 +1,337 @@
 const baseUrl = 'https://odk.server.com';
 const testServer = enableMockClient(baseUrl);
 
+const configuration = {
+  baseUrl,
+  email: 'a@b.com',
+  password: 'pass',
+};
+
+const mockResponse = (status, body) => ({
+  statusCode: status,
+  responseOptions: {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  },
+  data: body,
+});
+
+// Set up a fake auth endpoint
 before(() => {
   testServer
     .intercept({
       path: '/v1/sessions',
       method: 'POST',
     })
-    .reply(200, {
-      token: 'fake-token',
+    .reply(req => {
+      // Ensure credentials are correct and throw the appropriate response
+      const creds = JSON.parse(req.body);
+      if (
+        creds.email !== configuration.email ||
+        creds.password !== configuration.password
+      ) {
+        return mockResponse(401, {
+          message: 'Could not authenticate with the provided credentials.',
+          code: 401,
+        });
+      }
+      return mockResponse(200, {
+        token: 'fake-token',
+      });
     })
     .persist();
 });
 
-describe('Create Project', () => {
-  it('it should create a project', async () => {
+describe('getSubmissions', () => {
+  it('should fail if credentials are wrong', async () => {
+    testServer
+      .intercept({
+        path: '/v1/projects/1/forms/test_form.svc/Submissions',
+        method: 'GET',
+      })
+      .reply(200, fixtures.submissions);
+
+    const state = {
+      configuration: {
+        ...configuration,
+        password: 'wrooooon',
+      },
+    };
+
+    let error;
+    try {
+      await execute(getSubmissions(1, 'test_form'))(state);
+    } catch (e) {
+      error = e;
+    }
+
+    expect(error.statusCode).to.eql(401);
+    expect(error.statusMessage).to.eql('Unauthorized');
+    expect(error.body.message).to.eql(
+      'Could not authenticate with the provided credentials.'
+    );
+  });
+
+  it('should get form submissions', async () => {
+    testServer
+      .intercept({
+        path: '/v1/projects/1/forms/test_form.svc/Submissions',
+        method: 'GET',
+      })
+      .reply(200, fixtures.submissions);
+
+    const state = {
+      configuration,
+    };
+
+    // prettier-ignore
+    const finalState = await execute(
+      getSubmissions(1, 'test_form')
+    )(state);
+
+    expect(finalState.data).to.eql(fixtures.submissions.value);
+  });
+
+  it('should expand references', async () => {
+    testServer
+      .intercept({
+        path: '/v1/projects/1/forms/test_form.svc/Submissions',
+        method: 'GET',
+      })
+      .reply(200, fixtures.submissions);
+
+    const state = {
+      configuration,
+    };
+
+    // prettier-ignore
+    const finalState = await execute(
+      getSubmissions(() => 1, () => 'test_form')
+    )(state);
+
+    expect(finalState.data).to.eql(fixtures.submissions.value);
+  });
+
+  // TODO this isn't handled well yet
+  it.skip('it handles project not found', async () => {
```

Comment:
I'll decide in the morning whether I want to fix this or remove it

I want to check the behaviour now, out of the box, and if possible fix it in the runtime (which I need to do anyway). So it's a bit of a reminder.